src/precice/impl/GlobalDataContext.hpp

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <string>
+// #include "MappingContext.hpp"
+// #include "MeshContext.hpp"
+#include <Eigen/Core>
+#include "logging/Logger.hpp"
+#include "mesh/SharedPointer.hpp"
+#include "time/SharedPointer.hpp"
+#include "time/Time.hpp"
+
+namespace precice {
+
+namespace testing {
+// Forward declaration to friend the boost test struct
+class DataContextFixture;
+} // namespace testing
+//TODO: May need a GlobalDataContextFixture
+
+namespace impl {
+
+/**
+ * @brief Stores one Global (meshless) Data object.
+ *
+ * - GlobalDataContext is basically a stripped-down equivalent of DataContext.
+ *   Unlike DataContext, it has no associated mappings or meshes.
+ */
+class GlobalDataContext {
```

Comment:
Why does `GlobalDataContext` not inherit from `DataContext`? (compared to `ReadDataContext` and `WriteDataContext`)

Reply:
`DataContext` has a pointer `_mesh` and mesh-associated query functions which were not needed for global, so I created a new `GlobalDataContext` (just like `GlobalData` was created instead of reusing `Data`).

But now following the same philosophy of getting rid of `GlobalData` and making `Data` itself versatile, we can do the same here, i.e., reuse `DataContext` for global by for e.g. setting `_mesh` to nullptr in case of global contexts.

Reply:
> But now following the same philosophy of getting rid of `GlobalData` and making `Data` itself versatile, we can do the same here, i.e., reuse `DataContext` for global by for e.g. setting `_mesh` to nullptr in case of global contexts.

Sounds to me like a risky approach: We then get a general `Data` that is either implicitly global (has, for example, empty gradients and no `_mesh`) or lives on a mesh. To me this sounds like there should be an `AbstractData` which provides the base for `MeshData` and `GlobalData`. But not sure about all the implications this might have, though.

Reply:
For now I've made `Data` work for both mesh and global datas, and it looks good.
Regarding an `AabstractData`: The current `Data` class has no knowledge of any mesh (Datas don't point to any meshes. Only meshes point to their Datas).
Since `Data` is already dumb, it makes sense to me to use it as it is for both mesh-associated and global use cases.